chore(breaking-changes): add upcoming article with breaking changes (#618)
telerik/blazor-docs: upgrade/breaking-changes/3-0-0.md

---
title: 3.0.0
description: Handle the changes in the 3.0.0 release of the Telerik UI for Blazor components.
page_title: Breaking Changes in 3.0.0
slug: changes-in-3-0-0
position: 1
---

# Upcoming Breaking Changes in 3.0.0

> > The 3.0.0 release will be shipped in mid January. The following list of breaking changes will be updated with detailed explanation how to migrate your project.

Reviewer: Can we specify when we will provide the details (if before the release)?

```suggestion
> The 3.0.0 release will be shipped in mid January 2022. The following list of breaking changes will be updated with detailed explanation how to migrate your project.
```

## Component Changes

> - TileLayout - Introduce identifier of the TileLayout item. OnResize and OnReorder events will receive EventArgs that will point to the item that has been resized/reordered. This items comes from our [user voice](https://feedback.telerik.com/blazor/1489011)

Reviewer: User voice is a trademark of a competitor's product :)

```suggestion
- TileLayout - Introduce identifier of the TileLayout item. `OnResize` and `OnReorder` events will receive EventArgs that point to the resized/reordered item. This feature comes from our [public issue tracker](https://feedback.telerik.com/blazor/1489011)
```

> - TreeView - add `ExpandedItems` parameter to remove `ExpandedField`

Reviewer: ```suggestion
- TreeView - add `ExpandedItems` parameter and remove `ExpandedField`
```

> - Grid – change `OnRead` method to return the `Data`

Reviewer: This sounds like a major change. Is there anything else we can mention, even on high-level?

```suggestion
- Grid – change `OnRead` method to return the `Data`
```

- Grid – use `IFilterDescriptor` in GridState instead of `FilterDescriptorBase`
- DateInput – remove `ParsingErrorMessage` in favor of localization string for the error message

> - ContextMenu – add EventArgs to the OnClick event

Reviewer: ```suggestion
- ContextMenu – add EventArgs to the `OnClick` event
```

- All input elements – remove hardcoded width style in our Blazor components. In addition, the default style that comes from the themes will be updated to change the width to 100%.

> We are still designing changes in terms of editing by going through customer feedback. So, there might be changes coming from this direction as well.

Reviewer: I don't understand what this means - "in terms of editing"

## Common Changes

> - `IconName` class will be removed. All icon names should be retrieved from documentation.

Reviewer: ```suggestion
- [`IconName` class]({%slug common-kb-migration-from-iconname%}) (already obsolete) will be removed. All icon names should be retrieved from the [built-in icons documentation]({%slug general-information/font-icons%}#icons-list).
```

> - `ToODataString` extension method is moved to `Telerik.Blazor.Extensions` namespace.

Reviewer: ```suggestion
- `ToODataString` extension method is moved to the `Telerik.Blazor.Extensions` namespace.
```

## Parameter Naming

We are unifying our API with some parameter naming.

- `UpdateInterval` Parameter of Editor is changed to `DebounceDelay`
- `Content` RenderFragment Parameter of the Drawer is changed to `DrawerContent`

> - `IsCanceled` Parameter in StepperStepChangeEventArgs is renamed to `IsCancelled`

Reviewer: ```suggestion
- `IsCanceled` Parameter in `StepperStepChangeEventArgs` is renamed to `IsCancelled`
```

- `IsSeparatorField` is removed in favor of `SeparatorField`

## Rendering & Themes

- `k-widget` class is removed from the components
- `width:100%` is the new default width of all input-based components instead of `12.5em`

> - Html classes and dom elements hierarchy will be updated for Buttons, Inputs, Selects, DatePickers Checkbox, Switch. 

Reviewer: ```suggestion
- HTML classes and DOM element hierarchy will be updated for Buttons, Inputs, Selects, DatePickers Checkbox, Switch.
```
